services/user/Permissions/ui/src/permissions/App.tsx

```diff
@@ -1,83 +1,110 @@
 import { useEffect, useState } from "react";
 
 import { Button } from "@shadcn/button";
-import { Checkbox } from "@shadcn/checkbox";
-
-import { siblingUrl } from "@psibase/common-lib";
-import { Nav } from "@components/nav";
 
 import { supervisor } from "./perms_main";
+import { siblingUrl } from "@psibase/common-lib";
+import { ActivePermsOauthRequest } from "./db";
 
 export const App = () => {
     const thisServiceName = "permissions";
-    const [isLoading, setIsLoading] = useState<boolean>(true);
-    const [params, setParams] = useState<any>({});
-    const [isValidPermRequest, setIsValidPermRequest] = useState<any>({});
-
-    const init = async () => {
+    const [isLoading, setIsLoading] = useState(true);
+    const [validPermRequest, setValidPermRequest] = useState<any>(null);
+    const [error, setError] = useState<string | null>(null);
+    const initApp = async () => {
```

Comment:
I'd wrap `initApp` in a `useCallback()` so it's not re-created on each render (unless its dependencies change). Or alternatively, move it into the `useEffect()` right above where you call it. That would have a similar result.